### PR TITLE
fix: tags in subscriptions

### PR DESCRIPTION
**Pull request by im-adithya**
Previously we didn't support tags (mistakenly), this fix makes sure tags can be passed alongside the other filter attributes like this
```
{
  "ids": ["id1", "id2"],
  "kinds": [1,2],
  "authors": ["author1", "author2"],
  "since": 1721212121,
  "until": 1721212121,
  "limit": 10,
  "search": "example search",
  "#p": ["g3ta16yxxxxxxx"],
  "#e": ["g3ta16yxxxxxxx"]
}
```

### CODE REVIEW

**Comment by im-adithya** [comment]
Add README

File: internal/nostr/models.go [hunk 1]
```diff
@@ -153,13 +153,24 @@ type NIP47Request struct {
 	SignedEvent  *nostr.Event `json:"event"`
 }
 
+type Filter struct {
+	IDs     []string         `json:"ids,omitempty"`
+	Kinds   []int            `json:"kinds,omitempty"`
+	Authors []string         `json:"authors,omitempty"`
+	Tags    nostr.Tags       `json:"tags,omitempty"`
+	Since   *nostr.Timestamp `json:"since,omitempty"`
+	Until   *nostr.Timestamp `json:"until,omitempty"`
```
Comment (bumi) [inline internal/nostr/models.go:162]: should we use the nostr lib types there or our own types that we then pack into a nostr Filter?

**Comment by rolznz** [comment]
@im-adithya you have this "ADD README" comment. Can you update the filters in the readme or maybe even a full example with all the fields to be clear or add some examples? is this right for example?
```
{
    "relayUrl":"wss://relay.getalby.com/v1",
    "webhookUrl": "https://webhook.site/4adc457f-c1d1-4324-974c-0187d0f02872",
    "filter": {
        "kinds": [ 23196 ],
        "authors": [
            "c8986738660e5e5ee92e21a51e1f2e5915ad7ee9e972f301fc670f8eb47e9bed"
        ],
        "tags": [[
            "p",
            "8d5dcdf26d765ee1b4c279f01aefe337d1eb76cec93e7d276f9fa5b803079c43"
        ]]
    }
}
```

**Comment by bumi** [comment]
update the readme as mentioned above and we merge this.

**Comment by im-adithya** [comment]
Done, should be ready to merge!

**Comment by bumi** [comment]
so nothing changed?

**Comment by rolznz** [comment]
@im-adithya how does this work now? did you even test it?

**Comment by im-adithya** [comment]
Lol yes, nothing has to be changed. Just the way tags are supplied is changed (hence those big README changes). The easyjson internally handles passing the tags like this:
```
{
  "#p": ["g3ta16yxxxxxxx"],
  "#e": ["g3ta16yxxxxxxx"]
}
```
Checked everything and works fine! The tags get stored properly in the subscriptions table too.

**Pull request merged**